sample-apps/donet-sample-app/Controllers/MetricEmitter.cs

```diff
@@ -0,0 +1,137 @@
+                totaltimealiveMetricName = API_TOTAL_TIME_METRIC + "_" + instanceId;
+                totalHeapSizeMetricName = API_TOTAL_HEAP_SIZE + "_" + instanceId;
+                totalThreadsMetricName = API_TOTAL_THREAD_SIZE + "_" + instanceId;
+                cpuUsageMetricName = API_CPU_USAGE + "_" + instanceId;
+    
+            }
+            
+
+            meter.CreateObservableCounter(totalApiRequestSent,() => apiRequestSent, 
+                "1",
+                "Increments by one every time a sampleapp endpoint is used");
+
+            meter.CreateObservableCounter(totalApiBytesSentMetricName, () => totalBytesSent, 
+                "By",
+                "Keeps a sum of the total amount of bytes sent while the application is alive");
+
+            apiLatencyRecorder = meter.CreateHistogram<double>(latencyMetricName,
+                 "ms", 
+                 "Measures latency time in buckets of 100 300 and 500");
+
+            totalTimeSentObserver = meter.CreateCounter<long>(totaltimealiveMetricName,
+                "ms",
+                "Measures the total time the application has been alive");
+
+            meter.CreateObservableUpDownCounter(totalHeapSizeMetricName, () => totalHeapSize, 
+                "By",
+                "The current total heap size”");  
+
+            meter.CreateObservableCounter(totalThreadsMetricName, () => totalThreadSize,
+                "1",
+                "The current total number of threads”");
+
+            meter.CreateObservableGauge(cpuUsageMetricName, () => cpuUsage,
+                "1",
+                "Cpu usage percent");
+
+        }   
+        
+        public void emitReturnTimeMetric(long returnTime, String apiName, String statusCode) {
+            apiLatencyRecorder.Record(
+                returnTime,
+                new KeyValuePair<string, object>(DIMENSION_API_NAME, apiName),
+                new KeyValuePair<string, object>(DIMENSION_STATUS_CODE, statusCode));
+        }
+
+        public void apiRequestSentMetric(String apiName, String statusCode) {
+                apiRequestSent += 1;
+                Console.WriteLine("apiBs: "+ apiRequestSent);  
+        } 
+    
+        public void updateTotalBytesSentMetric(int bytes, String apiName, String statusCode) {
+            totalBytesSent += bytes;
+            Console.WriteLine("Total amount of bytes sent while the application is alive:"+ totalBytesSent);
+            apiNameValue = apiName;
+            statusCodeValue = statusCode;
+        }
+
+        public void updateTotalHeapSizeMetric(int heapSize) {
+            totalHeapSize += heapSize;
+        }
+
+        public void updateTotalThreadSizeMetric(int threadSize) {
+            totalThreadSize += threadSize;
+        }
+
+        public void updateCpuUsageMetric(int cpuUsage) {
+            cpuUsage = cpuUsage;
+        }
+
+        public void updateTotalTimeMetric(int totaltime) {
```

Comment:
This metric is never updated?

Reply:
Iant cpu metric supposed to be updated rather than getting updated

Reply:
you are right. That was not changing the` cpumetric`. Changed it